src/ui/pages/SwapForm/shared/useApproveHandler.ts

```diff
@@ -84,6 +86,7 @@ export function useApproveHandler({
       return { ...tx, gas: gasAsHex, gasLimit: gasAsHex };
     },
     staleTime: Infinity,
+    keepPreviousData: true,
```

Comment:
This is a bit controversial btw, I think it makes it possible to submit stale transaction

Reply:
I'm thinking about blocking the Send (Approve) button until the new/updated `approveTransaction` is constructed.
I mean literally: `<SignTransactionButton ... disabled={... || allowanceIsLoading}`

Reply:
✅